python/tskit/trees.py

```diff
@@ -1490,10 +1490,78 @@ def _levelorder_traversal(self, u):
             yield v
 
     def _timeasc_traversal(self, u):
-        yield from sorted(self.nodes(u, order="levelorder"), key=self.time)
+        """Sorts by increasing time but falls back to increasing ID for equal times.
+        """
+        time_id_pairs = [
+            [self.time(node), node] for node in self.nodes(u, order="levelorder")
+        ]
+        time_id_pairs.sort()
+        for pair in time_id_pairs:
+            yield pair[1]
 
     def _timedesc_traversal(self, u):
-        yield from sorted(self.nodes(u, order="levelorder"), key=self.time, reverse=True)
+        """Sorts by decreasing time but falls back to decreasing ID for equal times.
+        """
+        time_id_pairs = [
+            [self.time(node), node] for node in self.nodes(u, order="levelorder")
+        ]
+        time_id_pairs.sort(reverse=True)
+        for pair in time_id_pairs:
+            yield pair[1]
+
+    def _minlex_postorder_traversal(self, u):
+        """Postorder traversal that visits leaves in minimum lexicographic order.
+
+        Minlex stands for minimum lexicographic. We wish to visit a tree in such
+        a way that the leaves visited, when their IDs are listed out, have
+        minimum lexicographic order. This is a useful ordering for drawing
+        multiple Trees of a TreeSequence, as it leads to more consistency
+        between adjacent Trees.
+        """
+        stack = collections.deque([u])
+        parent = NULL
+        # For perf we store these to avoid lookups in the tight loop
+        pop = stack.pop
+        extend = stack.extend
+        get_children = self.children
+        get_parent = self.get_parent
+
+        # We store a dictionary from internal node ID to min leaf under the node
+        # First we compute this dictionary using a postorder traversal
+        min_leaf_dict = {}
```

Comment:
No, we can't compute it using `nodes(order="postorder")`, because the postorder underneath also needs to be minlex. However, I have gotten rid of the perf optimisations and added lots of comments. I forgot a lot of what the code was doing as well!

Reply:
If we want to make things even more readable, there is a recursive solution in tests/__init__.py, but I think we should still stick to an iterative solution here.